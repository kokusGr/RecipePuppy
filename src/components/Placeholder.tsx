import React from 'react'
import { View, Text, StyleSheet } from 'react-native'

interface Props {
  query: string
}

const Placeholder = ({ query }: Props) => {

  return (
    <View style={styles.root}>
      <Text style={styles.heading}>We couldn't find any recipes matching these ingredients: </Text>
      <Text style={styles.query}>{query}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 60,
  },
  heading: {
    fontSize: 22,
    textAlign: 'center',
    paddingHorizontal: 16,
    marginBottom: 16,
  },
  query: {
    fontSize: 18,
    textAlign: 'center',
    fontStyle: 'italic',
    paddingHorizontal: 16,
  },
})

export default Placeholder
