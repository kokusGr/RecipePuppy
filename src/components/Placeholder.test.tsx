import React from 'react'
import renderer from 'react-test-renderer'

import Placeholder from './Placeholder'

it('matches snapshot', () => {
    const tree = renderer.create(<Placeholder query="test" />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders message with query', () => {
    const tree = renderer.create(<Placeholder query="test" />)
    expect(tree.root.findByProps({ testID: 'query' }).props.children).toEqual('test')
})
