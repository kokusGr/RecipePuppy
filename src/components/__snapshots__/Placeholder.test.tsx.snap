// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 60,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 22,
        "marginBottom": 16,
        "paddingHorizontal": 16,
        "textAlign": "center",
      }
    }
  >
    We couldn't find any recipes matching these ingredients: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontStyle": "italic",
        "paddingHorizontal": 16,
        "textAlign": "center",
      }
    }
    testID="query"
  >
    test
  </Text>
</View>
`;
